SofaKernel/modules/SofaBaseTopology/PointSetTopologyContainer.cpp

```diff
@@ -33,6 +36,25 @@ namespace component
 namespace topology
 {
 
+namespace
+{
+
+struct GeneratePointID
+{
+    typedef sofa::core::topology::BaseMeshTopology::PointID PointID;
+
+    GeneratePointID( PointID startId = PointID(0) )
+    :current(startId)
+    {
+    }
+
+    PointID operator() () { return current++; }
+
+    PointID current;
+};
```

Comment:
This is probably in order to be able to use std::generate() instead of a manual loop construct. Note that these changes were made before the decision to allow for c++11 constructs, it could potentially be replaced by a functor now.